### DISCUSSION TITLE
Keyboard and mouse don't work on emulated MacOS (x86_64)

### DISCUSSION THREAD

**Discussion by qu0dy** [utmapp/UTM#4704] [UTM Help (macOS)] [answered]
**Describe the issue**  
Keyboard and mouse don't work after kernel launch (https://khronokernel.github.io/apple/silicon/2021/01/17/QEMU-AS.html or https://www.nicksherlock.com/2021/06/installing-macos-12-monterey-on-proxmox/ guides).

**Configuration**  
* UTM Version: v4.1.0 or v4.0.9
* host macOS Version: 13.1 Beta
* guest macOS Version: 
    * Big Sur — mouse/keyboard don't work
    * Monterey
        * don't work
            * AppleUSBHostPort::createDevice: failed to create device (0xe00002bc) with `-global nec-usb-xhci.msi=off`
            * `Assertion failed: (vector < nr_vectors), function msi_notify, file msi.c, line 358.` without `-global nec-usb-xhci.msi=off`
        * do work
            * Maximum shared USB device == 0
            * Share USB devices from host is off
           * Force PS/2 controller
           * `-usb -device usb-kbd -device usb-tablet`
    * Ventura — mouse seems to work, but on a frozen loading screen.
* Mac Chip: M1 Pro
* EFI:
    * https://github.com/khronokernel/khronokernel.github.io/blob/master/Binaries/OpenCore/EFI-MODERN.img.zip?raw=true
    * https://github.com/thenickdude/KVM-Opencore/releases/download/v19/OpenCore-v19.iso.gz

**Comment by osy** [comment, marked as answer]
You need to capture the cursor if the usb tablet device isn't working correctly on the guest (especially if you force ps/2 connection). For macos, you can try this https://github.com/utmapp/UTM/issues/2610#issuecomment-901027661 which you can add in QEMU options as an additional argument.
